### PR TITLE
 #1199 : Compare with six.integer_types instead of (int, six.PY2 and long)

### PR DESCRIPTION
**Pull request by khchine5**

### CODE REVIEW

**Comment by khchine5** [comment]
Any particular problem with this ?

**Comment by lsaffre** [comment]
I wondered how to handle the case when limit is None. Your change is actually correct, but it seems that the code was wrong before: when limit is None, we actually do want a phantom row. Can you check whether the tests pass when you do as in my remark?

File: lino/modlib/extjs/views.py [hunk 1]
```diff
@@ -608,12 +606,12 @@ def get(self, request, app_label=None, actor=None):
             total_count = ar.get_total_count()
             # raise Exception("20171208 {}".format(ar.data_iterator.query))
             for row in ar.create_phantom_rows():
-                if len(rows)+1 < ar.limit\
-                        or ar.limit == total_count + 1:
+                if (ar.limit is not None and len(rows) + 1 < ar.limit) or (
+                                ar.limit is None and isinstance(row, PhantomRow)) or ar.limit == total_count + 1:
```
Comment (khchine5) [inline lino/modlib/extjs/views.py]: Add a check if the limit is None and the row is a PhantomRow so we actually append the row.

**Comment by lsaffre** [comment]
Why so complicated? Why not my suggestion?

**Pull request merged**